airflow/jobs/scheduler_job_runner.py

```diff
@@ -721,6 +721,7 @@ def _process_executor_events(self, session: Session) -> int:
             if state == TaskInstanceState.QUEUED:
                 ti.external_executor_id = info
                 self.log.info("Setting external_id for %s to %s", ti, info)
+                session.commit()
```

Comment:
I'm not a DB/SQLA expert, but do we want to commit here, our outside of the for loop? Here is going to do a commit for each queued task individually that we got from the query. Which could be many? Interested to hear others weigh in.

Reply:
A valid point, and I don't know the right answer.   I'll see what others think.  It may be safest to move it outside the loop